packages/medusa/src/services/order-edit.ts

```diff
@@ -680,6 +680,39 @@ export default class OrderEditService extends TransactionBaseService {
     })
   }
 
+  async complete(
+    orderEditId: string,
+    context: { loggedInUserId?: string }
+  ): Promise<OrderEdit> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEdit = await this.retrieve(orderEditId)
+
+      if (orderEdit.status === OrderEditStatus.CONFIRMED) {
+        return orderEdit
+      }
+
+      if (orderEdit.status !== OrderEditStatus.REQUESTED) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Cannot complete an order edit with status ${orderEdit.status}`
+        )
+      }
+
+      // TODO once payment collection is done
+      /*const paymentCollection = await this.paymentCollectionService_.withTransaction(manager).retrieve(orderEdit.payment_collection_id)
+      if (!paymentCollection.authorized_at) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "Unable to complete an order edit if the payment is not authorized"
+        )
+      }*/
```

Comment:
thought: would eliminate the payment collection dependency and instead do this check in the request handler instead.

Reply:
it would means that the order edit status will be also check in the handler? cause we don't want to throw on an already confirmed edit for example.

Reply:
Makes sense 🤘